builtin/providers/dockercloud/resource_dockercloud_node_cluster.go

```diff
@@ -0,0 +1,255 @@
+			},
+			"tags": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceDockercloudNodeClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	provider := d.Get("node_provider").(string)
+	region := d.Get("region").(string)
+	size := d.Get("size").(string)
+
+	opts := dockercloud.NodeCreateRequest{
+		Name:     d.Get("name").(string),
+		Region:   fmt.Sprintf("/api/infra/v1/region/%s/%s/", provider, region),
+		NodeType: fmt.Sprintf("/api/infra/v1/nodetype/%s/%s/", provider, size),
+	}
+
+	if attr, ok := d.GetOk("disk"); ok {
+		opts.Disk = attr.(int)
+	}
+
+	if attr, ok := d.GetOk("node_count"); ok {
+		opts.Target_num_nodes = attr.(int)
+	}
+
+	tags := d.Get("tags.#").(int)
+	if tags > 0 {
+		opts.Tags = make([]dockercloud.NodeTag, 0, tags)
+		for i := 0; i < tags; i++ {
+			key := fmt.Sprintf("tags.%d", i)
+			opts.Tags = append(opts.Tags, dockercloud.NodeTag{Name: d.Get(key).(string)})
+		}
+	}
+
+	nodeCluster, err := dockercloud.CreateNodeCluster(opts)
+	if err != nil {
+		return err
+	}
+
+	if err = nodeCluster.Deploy(); err != nil {
+		return fmt.Errorf("Error creating node cluster: %s", err)
+	}
+
+	d.SetId(nodeCluster.Uuid)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:        []string{"Deploying"},
+		Target:         []string{"Deployed"},
+		Refresh:        newNodeClusterStateRefreshFunc(d, meta),
+		Timeout:        60 * time.Minute,
+		Delay:          10 * time.Second,
+		MinTimeout:     3 * time.Second,
+		NotFoundChecks: 60,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for node cluster (%s) to become ready: %s", d.Id(), err)
+	}
+
+	return resourceDockercloudNodeClusterRead(d, meta)
+}
+
+func resourceDockercloudNodeClusterRead(d *schema.ResourceData, meta interface{}) error {
+	nodeCluster, err := dockercloud.GetNodeCluster(d.Id())
+	if err != nil {
+		if strings.Contains(err.Error(), "404 NOT FOUND") {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving node cluster: %s", err)
+	}
+
+	if nodeCluster.State == "Terminated" {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("name", nodeCluster.Name)
+	d.Set("node_count", nodeCluster.Target_num_nodes)
+	d.Set("disk", nodeCluster.Disk)
+	d.Set("tags", flattenNodeTags(nodeCluster.Tags))
+
+	return nil
+}
+
+func resourceDockercloudNodeClusterUpdate(d *schema.ResourceData, meta interface{}) error {
+	var opts dockercloud.NodeCreateRequest
+
+	if d.HasChange("node_count") {
+		_, newNum := d.GetChange("node_count")
+		opts.Target_num_nodes = newNum.(int)
+	}
+
+	if d.HasChange("tags") {
+		_, newTags := d.GetChange("tags")
+		tags := newTags.([]interface{})
+		opts.Tags = make([]dockercloud.NodeTag, 0, len(tags))
+
+		for _, tag := range tags {
+			opts.Tags = append(opts.Tags, dockercloud.NodeTag{Name: tag.(string)})
+		}
+	}
+
+	nodeCluster, err := dockercloud.GetNodeCluster(d.Id())
```

Comment:
`GetNodeCluster` returns a NodeCluster type which has the `Update` method. They reference an internal `self.Uuid` for updating (https://github.com/docker/go-dockercloud/blob/0076cacbd4123217c1fae6844f13d7cbd6f7a466/dockercloud/node_cluster.go#L140).